pyvrp/cpp/educate/Route.cpp

```diff
@@ -136,7 +136,7 @@ void Route::update()
     isLoadFeasible_ = static_cast<size_t>(load_) <= data.vehicleCapacity();
 
     timeWarp_ = nodes.back()->twBefore.totalTimeWarp();
-    isTimeWarpFeasible_ = timeWarp_ == 0;
+    isTimeWarpFeasible_ = equal(timeWarp_, duration_type(0));
```

Comment:
I'm not sure if we want/need a tolerance here? Doubles can represent exact 0 and the timewarp will be zero if all arrivals are before the end of the time windows. I'm not sure if we need a tolerance there

Reply:
When both are exactly zero the comparison is exact; see the implementation of `equal()`. 

A problem would be when an arrival happens at about the end of a time window, say at `1.00001` for a time window of `[0, 1]`, we'd have `0.00001` time warp. I think that's just a rounding error, but the route would then not be considered time warp feasible because that value is not zero. To avoid such cases I think we should have a tolerance here.

Reply:
Ok, I agree that is a problem if we use doubles to represent integer distances and time windows, which is quite likely. If we actually compute distances using double precision, I think it makes more sense to put the boundary at `1.00000` as you will not exactly arrive at this time anyway, but it does not hurt either so let's keep it this way.